diff --git a/cookbooks/tomcat/recipes/default.rb b/cookbooks/tomcat/recipes/default.rb
index b1a38bc..02e5868 100644
--- a/cookbooks/tomcat/recipes/default.rb
+++ b/cookbooks/tomcat/recipes/default.rb
@@ -1,14 +1,19 @@
+service 'tomcat7' do
+  action :stop
+end
 apt_update
-apt_package 'tomcat7'
-apt_package 'tomcat7-admin'
-file '/etc/tomcat7/tomcat-users.xml' do
+apt_package 'libtomcat6-java'
+apt_package 'tomcat6-common'
+apt_package 'tomcat6'
+apt_package 'tomcat6-admin'
+file '/etc/tomcat6/tomcat-users.xml' do
   content %{<?xml version='1.0' encoding='utf-8'?>
 <tomcat-users>
   <role rolename="manager-script"/>
   <user username="admin" password="admin" roles="manager-script"/>
 </tomcat-users>}
 end
-service 'tomcat7' do
+service 'tomcat6' do
   action :start
 end
-execute 'chown -R tomcat7:tomcat7 /var/lib/tomcat7/webapps'
+execute 'chown -R tomcat6:tomcat6 /var/lib/tomcat6/webapps'
